tq/meter.go

```diff
@@ -163,6 +163,19 @@ func (m *Meter) FinishTransfer(name string) {
 	m.fileIndexMutex.Unlock()
 }
 
+// Flush sends the latest progress update, while leaving the meter active.
+func (m *Meter) Flush() {
+	if m == nil || m.skipUpdate() {
+		return
+	}
+
+	m.updates <- &tasklog.Update{
```

Comment:
I think that one thing we could do to limit the number of places we do a channel write explicitly is to teach `update()` to take a `force bool` as a formal argument, and then pass that inside the body of the struct, like:

```go
func (m *Meter) update(force bool) { 
        m.updates <- &tasklog.Update{Force: true, ...}
}
```

I don't mind the duplication, but I think that channel writes in general can be fairly nuanced, so the fewer the better.